### PR TITLE
Form disable meta

### PR DESCRIPTION
**Pull request by richardhallett**
For #132

### CODE REVIEW

File: app/components/doi-form.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import Component from '@ember/component';
+
+export default Component.extend({
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+
+    if (this.model.get("source") == "fabricaForm") {
+      this.set('metadataFieldsDisabled', false);
```
Comment (mfenner) [inline app/components/doi-form.js]: You could make this the default in line 4.

**Comment by richardhallett** [comment]
> Why is there a `doi-form` component. It seems to add complexity.

Because we need specific functionality that applies to the form, this seemed a logical way to encapsulate the behaviour of the metadata form fields separate to that of other parts.
Do you have another suggestion of where to put the logic around deciding if the form elements should be disabled or not?

**Comment by mfenner** [comment]
@richardhallett the form is displayed in four slightly different versions: new/edit and form/file upload. We need to disable the form in only one of them (edit form). So it would make sense for me to put that logic into the template for `edit form`: https://github.com/datacite/bracco/blob/master/app/templates/clients/show/dois/show/edit.hbs

**Comment by richardhallett** [comment]
> @richardhallett the form is displayed in four slightly different versions: new/edit and form/file upload. We need to disable the form in only one of them (edit form). So it would make sense for me to put that logic into the template for `edit form`: https://github.com/datacite/bracco/blob/master/app/templates/clients/show/dois/show/edit.hbs

The usage of the field could be in that I suppose, but given I needed to set the metadataFieldsDisabled this seemed best to put it in component code which is where the backing of doi-form.js comes in? Where instead would you put that in place, create a backend component logic for the show/edit.js?

File: app/controllers/clients/show/dois/new.js [hunk 2]
```diff
@@ -23,6 +23,8 @@ export default Controller.extend({
         doi.set('descriptions', [{ description: doi.get('descriptions'), descriptionType: 'Abstract' }]);
       }
 
+      doi.set("source", "fabricaForm");
```
Comment (mfenner) [inline app/controllers/clients/show/dois/new.js]: @richardhallett you can also do this directly when you create the DOI, i.e. start with defaults.

**Pull request merged**